runtime/contract_update_test.go

```diff
@@ -75,24 +81,33 @@ func TestContractUpdateWithDependencies(t *testing.T) {
 		decodeArgument: func(b []byte, t cadence.Type) (value cadence.Value, err error) {
 			return json.Decode(nil, b)
 		},
-		getProgram: func(location Location) (*interpreter.Program, error) {
+		getAndSetProgram: func(
+			location Location,
+			load func() (*interpreter.Program, error),
+		) (
+			program *interpreter.Program,
+			err error,
+		) {
 			_, isTransactionLocation := location.(common.TransactionLocation)
-			if checkGetSetProgram && !isTransactionLocation {
+			if checkGetAndSetProgram && !isTransactionLocation {
 				require.Equal(t, location, fooLocation)
 				require.False(t, getProgramCalled)
-				getProgramCalled = true
 			}
-			// Always force to get the old program from the source during the update.
-			return nil, nil
```

Comment:
This was an incorrect implementation of `GetProgram`. Instead, it should return what it loaded. Also, programs are only effective during the duration of execution (e.g. of a transaction or script), not across them, so clear all programs. This is because contract updates are delayed, and the test runtime interface does not properly commit/overwrite the new programs